### PR TITLE
Adding support for ARRAY_SORT_DESC to Presto

### PR DESCRIPTION
**Pull request by jainavi17**
Adding an UDF for sorting array elements in the descending order. The elements of array must be orderable. Null elements will be placed at the end of the returned array.

array_sort_desc(array(T)) -> array(T)

Test plan
Added unit tests.
Build successfully using the following terminal command
- `mvn -Dtest=TestArraySqlFunctions install`

```
== RELEASE NOTES ==

General Changes
* Add :func:`array_sort_desc` function to sort an array in the descending order
```

### CODE REVIEW

**Comment by kaikalur** [comment]
> Adding an UDF for sorting array elements in the descending order. The elements of array must be orderable. Null elements will be placed at the end of the returned array.
> 
> array_sort_desc(array(T)) -> array(T)
> 
> Test plan Added unit tests. Build successfully using the following terminal command
> 
> * `mvn -Dtest=TestArraySqlFunctions install`
> 
> ```
> == RELEASE NOTES ==
> 
> General Changes
> * Adding support for ARRAY_SORT_DESC
> * Returns an array sorted in the descending order. The elements should be orderable else function will throw an error. Similar to ARRAY_SORT, null elements are placed at the end of the array.
> 
> .. function:: array_sort_desc(array(T)) -> array(T)
>    Returns sorted ``array`` in descending order
> 
> `SELECT array_sort_desc(ARRAY [100, 1, 10, 50]); -- [100, 50, 10, 1]`
> `SELECT array_sort_desc(ARRAY [null, 100, null, 1, 10, 50]); -- [100, 50, 10, 1, null, null]`
> `SELECT array_sort_desc(ARRAY [null, 100, null, 1, 10, 50]); -- [100, 50, 10, 1, null, null]`
> ```

Formatting release notes

**Review by kaikalur — APPROVED** [review]

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java [hunk 1]
```diff
@@ -214,4 +217,28 @@ public void testArrayMinBy()
...
+        assertFunction("ARRAY_SORT_DESC(ARRAY [double'1.0', double'2.0'])", new ArrayType(DOUBLE), ImmutableList.of(2.0d, 1.0d));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [double'1.0', double'2.0'])", new ArrayType(DOUBLE), ImmutableList.of(2.0d, 1.0d));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [null, double'-3.0', double'2.0', null])", new ArrayType(DOUBLE), asList(2.0d, -3.0d, null, null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY ['a', 'bb', 'c'])", new ArrayType(createVarcharType(2)), ImmutableList.of("c", "bb", "a"));
+        assertFunction("ARRAY_SORT_DESC(ARRAY ['a', 'bb', 'c', null])", new ArrayType(createVarcharType(2)), asList("c", "bb", "a", null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [null, null, null])", new ArrayType(UNKNOWN), asList(null, null, null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [])", new ArrayType(UNKNOWN), emptyList());
+        assertFunction("ARRAY_SORT_DESC(null)", new ArrayType(UNKNOWN), null);
+        assertFunction("ARRAY_SORT_DESC(ARRAY [ARRAY['a'], ARRAY['b', 'b'], ARRAY['c']])",
```
Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java:234]: put the first param to its own line

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java [hunk 2]
```diff
@@ -214,4 +217,28 @@ public void testArrayMinBy()
...
+                new ArrayType(new ArrayType(createVarcharType(1))),
+                ImmutableList.of(
+                        ImmutableList.of("c"), ImmutableList.of("b", "b"), ImmutableList.of("a")));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [ARRAY['a'], ARRAY['b', 'b'], ARRAY['c'], null, null, ARRAY['a', NULL]])",
```
Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java:238]: put the first param to its own line

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java [hunk 3]
```diff
@@ -214,4 +217,28 @@ public void testArrayMinBy()
...
+        assertFunction("ARRAY_SORT_DESC(ARRAY [null, double'-3.0', double'2.0', null])", new ArrayType(DOUBLE), asList(2.0d, -3.0d, null, null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY ['a', 'bb', 'c'])", new ArrayType(createVarcharType(2)), ImmutableList.of("c", "bb", "a"));
+        assertFunction("ARRAY_SORT_DESC(ARRAY ['a', 'bb', 'c', null])", new ArrayType(createVarcharType(2)), asList("c", "bb", "a", null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [null, null, null])", new ArrayType(UNKNOWN), asList(null, null, null));
+        assertFunction("ARRAY_SORT_DESC(ARRAY [])", new ArrayType(UNKNOWN), emptyList());
+        assertFunction("ARRAY_SORT_DESC(null)", new ArrayType(UNKNOWN), null);
+        assertFunction("ARRAY_SORT_DESC(ARRAY [ARRAY['a'], ARRAY['b', 'b'], ARRAY['c']])",
+                new ArrayType(new ArrayType(createVarcharType(1))),
+                ImmutableList.of(
+                        ImmutableList.of("c"), ImmutableList.of("b", "b"), ImmutableList.of("a")));
```
Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestArraySqlFunctions.java:237]: we can put these two lines into one

**Review by highker — APPROVED** [review]
The release note is too verbose. We only need one (long) line to explain what it does in English. Also, we need `:func:` annotation for the function name so it can be linked to the doc you wrote.

**Comment by v-jizhang** [comment]
@bot kick off tests

**Comment by jainavi17** [comment]
> The release note is too verbose. We only need one (long) line to explain what it does in English. Also, we need `:func:` annotation for the function name so it can be linked to the doc you wrote.

Done!

**Comment by jainavi17** [comment]
Can we merge this PR?

**Pull request merged**